pymongo/monitor.py

```diff
@@ -255,7 +261,17 @@ def _check_once(self) -> ServerDescription:
         address = self._server_description.address
         if self._publish:
             assert self._listeners is not None
-            self._listeners.publish_server_heartbeat_started(address)
+            sd = self._server_description
+            # TODO: this is not accurate in the rare case the pool has a
```

Comment:
Still TODO?   Needs a follow up ticket?

Reply:
I don't think fixing this is worth the effort, especially since it should be so rare (in streaming the socket is almost always checked out of the pool so the connection would need to become stale in the short window between the check in and the checkout) and the impact of this is trivial (ServerHeartbeatStartedEvent.awaited will be True instead of False).